### PR TITLE
UK-LON-X1

### PR DESCRIPTION
**Pull request by oli-barker**

### CODE REVIEW

**Comment by oli-barker** [comment]
Setting up new building please ignore

**Comment by charbull** [comment]
Hi @oli-barker can you please sign the cla

Thank you

File: ontology/yaml/resources/fields/telemetry_fields.yaml [hunk 1]
```diff
@@ -772,3 +772,10 @@ literals:
 - zone_air_occupied_heating_temperature_setpoint
 - mixed_air_relative_humidity_sensor
 - mixed_air_dewpoint_temperature_sensor
+
+
+# Added for UK-LON-X1
```
Comment (charbull) [inline ontology/yaml/resources/fields/telemetry_fields.yaml:777]: can you please remove this tag (UK-LON-X1)

**Comment by oli-barker** [comment]
> @googlebot I signed it

**Comment by oli-barker** [comment]
> @googlebot I signed it!

File: ontology/yaml/resources/HVAC/entity_types/GENERALTYPES.yaml [hunk 2]
```diff
@@ -18,6 +18,17 @@
 ####      See model_hvac.md for detailed definitions      ####
 ##############################################################
 
+
+TRU:
```
Comment (tasodorff) [inline ontology/yaml/resources/HVAC/entity_types/GENERALTYPES.yaml:22]: Doesn't match BOS v2.3 standard. Fix this.

File: ontology/yaml/resources/HVAC/entity_types/GENERALTYPES.yaml [hunk 3]
```diff
@@ -18,6 +18,17 @@
...
+  id: "oli0001"
```
Comment (tasodorff) [inline ontology/yaml/resources/HVAC/entity_types/GENERALTYPES.yaml:23]: Omit all IDs, the system will append these if needed.

File: ontology/yaml/resources/HVAC/entity_types/GENERALTYPES.yaml [hunk 4]
```diff
@@ -18,6 +18,17 @@
...
+  description: "Tag for trench heating units. Trench Heating units are in floor airconditioning units.
```
Comment (tasodorff) [inline ontology/yaml/resources/HVAC/entity_types/GENERALTYPES.yaml:24]: Clarify "particular to under-floor air distribution systems" or something similar.

File: ontology/yaml/resources/HVAC/entity_types/TRU.yaml [hunk 5]
```diff
@@ -0,0 +1,30 @@
+# Trench Heating File for UK-LON-X1 project
+
+########################
+### Canonical Types  ###
+########################
+
+# This file defines Trench heating unit types.
+
+TRH_DFSS_DFVSC_DSP_CHWZC:
```
Comment (tasodorff) [inline ontology/yaml/resources/HVAC/entity_types/TRU.yaml:9]: This name doesn't match the types implemented below. These should match by convention.

File: ontology/yaml/resources/HVAC/entity_types/TRU.yaml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  id: "oli0002"
+  description: "Typical trench heater with cooling-only"
+  is_canonical: true
+  implements:
+  - TRU
```
Comment (tasodorff) [inline ontology/yaml/resources/HVAC/entity_types/TRU.yaml:14]: fix

File: ontology/yaml/resources/HVAC/entity_types/TRU.yaml [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+  - DFVSC
```
Comment (tasodorff) [inline ontology/yaml/resources/HVAC/entity_types/TRU.yaml:15]: Please also implement DFSS

File: ontology/yaml/resources/HVAC/entity_types/TRU.yaml [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+  - CHWZTC
+  - RMM
+
+TRH_DFSS_DFVSC_DSP_HWZC_CHWZC:
+  id: "oli0003"
+  description: "Typical trench heater with cooling and heating"
+  is_canonical: true
+  implements:
+  - TRU
+  - DFVSC
```
Comment (tasodorff) [inline ontology/yaml/resources/HVAC/entity_types/TRU.yaml:26]: Same note as above; please implement DFSS also

File: ontology/yaml/resources/HVAC/entity_types/TRU.yaml [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+  description: "Typical trench heater with cooling-only"
+  is_canonical: true
+  implements:
+  - TRU
+  - DFVSC
+  - ZTC
+  - CHWZTC
+  - RMM
+
+TRH_DFSS_DFVSC_DSP_HWZC_CHWZC:
```
Comment (tasodorff) [inline ontology/yaml/resources/HVAC/entity_types/TRU.yaml:20]: Same note as above; match implemented types.

File: ontology/yaml/resources/METERS/entity_types/HM.yaml [hunk 10]
```diff
@@ -24,4 +24,17 @@ HM_CHW:
   - chilled_supply_water_temperature_sensor
   - chilled_return_water_temperature_sensor
   - cooling_thermal_power_sensor
-  - cooling_thermal_energy_accumulator
\ No newline at end of file
+  - cooling_thermal_energy_accumulator
+
+HM_HWS:  
+  id: "oli0004"
```
Comment (tasodorff) [inline ontology/yaml/resources/METERS/entity_types/HM.yaml:30]: reminder to remove all fake IDs

**Review by tasodorff — CHANGES_REQUESTED** [review]
Main comment is to change trench heater abbreviation to match BOS standard. Several minor fixes (remove all IDs, add a few requested abstract types, type name should match concatenated 'implements' types by convention, etc.).

**Comment by tasodorff** [comment]
I still don't see the type names updated in your latest PR.

**Comment by oli-barker** [comment]
> I still don't see the type names updated in your latest PR.

I updated the TRU entry in general types to TRH in line with BOS 2.3. Also changed the TRU.yaml file to TRH.yaml and defined 2 types of trench heater. One with just heating and one with both. I haven't defined TRHC as its effectively a subtype of the main item. This seemed the same in line with FCU where it can be any configuration of heating and cooling etc but is still an FCU.

File: ontology/yaml/resources/HVAC/entity_types/AHU.yaml [hunk 11]
```diff
@@ -3566,4 +3566,52 @@ AHU_US_SVL_MAT2_4:
   - supply_fan_run_command
   - supply_fan_speed_frequency_sensor
   - supply_fan_speed_percentage_command
-  - supply_air_flowrate_sensor
\ No newline at end of file
+  - supply_air_flowrate_sensor
+  
+  
+AHU_UK_LON_X1_1:
```
Comment (tasodorff) [inline ontology/yaml/resources/HVAC/entity_types/AHU.yaml:3572]: A few notes:
1. Why dont we have any information for the supply/exhaust fans (command/status/speeds), orr any outside air information?
2. Do not use the name of the site in the type name.

File: ontology/yaml/resources/HVAC/entity_types/AHU.yaml [hunk 12]
```diff
@@ -3566,4 +3566,52 @@ AHU_US_SVL_MAT2_4:
...
+  - frost_coil_status
+  - cooling_coil_percentage_command
+  - cooling_coil_status
+  - master_alarm
+  - exhaust_fan_speed_percentage_sensor
+  - exhaust_air_flowrate_sensor
+  - supply_air_flowrate_sensor
+  - return_air_temperature_sensor  
+  
+AHU_UK_LON_X1_2:
```
Comment (tasodorff) [inline ontology/yaml/resources/HVAC/entity_types/AHU.yaml:3590]: Same note as above -- remove Google specific name.

File: ontology/yaml/resources/fields/telemetry_fields.yaml [hunk 13]
```diff
@@ -772,3 +772,38 @@ literals:
...
 - mixed_air_dewpoint_temperature_sensor
+
+- heating_supply_water_temperature_sensor
+- heating_return_water_temperature_sensor
+- heating_thermal_power_sensor
+- heating_thermal_energy_accumulator
+
+#added for UK-LON-X1
+
+- heating_wheel_percentage_sensor
```
Comment (tasodorff) [inline ontology/yaml/resources/fields/telemetry_fields.yaml:783]: heating wheel what percentage sensor? 

percent of air flowing through it? speed at which its rotating? percent capacity? needs more specificity.

File: ontology/yaml/resources/fields/telemetry_fields.yaml [hunk 14]
```diff
@@ -772,3 +772,38 @@ literals:
...
+- heat_recovery_damper_percentage_sensor
+- discharge_air_filter_differential_pressure_sensor
+- discharge_air_filter_differential_pressure_setpoint
+- return_air_filter_differential_pressure_sensor
+- return_air_filter_differential_pressure_setpoint
+- cooling_coil_percentage_command
```
Comment (tasodorff) [inline ontology/yaml/resources/fields/telemetry_fields.yaml:789]: is this the valve?

File: ontology/yaml/resources/fields/telemetry_fields.yaml [hunk 15]
```diff
@@ -772,3 +772,38 @@ literals:
...
+- cooling_coil_status:
+  - ACTIVE
+  - INACTIVE
+- heating_coil_percentage_command
```
Comment (tasodorff) [inline ontology/yaml/resources/fields/telemetry_fields.yaml:793]: why are you using this and not heating_water_valve_percentage command? is it not hydronic? if not, what is it?

File: ontology/yaml/resources/fields/telemetry_fields.yaml [hunk 16]
```diff
@@ -772,3 +772,38 @@ literals:
...
+- frost_coil_status:
```
Comment (tasodorff) [inline ontology/yaml/resources/fields/telemetry_fields.yaml:794]: again, what is this a status of? flow through coil?

File: ontology/yaml/resources/fields/telemetry_fields.yaml [hunk 17]
```diff
@@ -772,3 +772,38 @@ literals:
...
+  - ACTIVE
+  - INACTIVE
+- max_supply_air_temperature_setpoint
+- min_supply_air_temperature_setpoint
```
Comment (tasodorff) [inline ontology/yaml/resources/fields/telemetry_fields.yaml:798]: same as above; use `low_limit` if not an aggregation.

**Review by tasodorff — CHANGES_REQUESTED** [review]
Clarifications needed on some definitions, and alternate names required.

**Comment by tasodorff** [comment]
@oli-barker what is the status of these? please make changes or the PR will be closed without committing the changes.

**Comment by tasodorff** [comment]
@oli-barker status?

**Comment by tasodorff** [comment]
Closed due to inactivity. Can reopen if necessary.

**Pull request closed without merging**